projects/plugins/jetpack/modules/masterbar/admin-menu/load.php

```diff
@@ -7,10 +7,57 @@
 
 namespace Automattic\Jetpack\Dashboard_Customizations;
 
-if ( jetpack_is_atomic_site() ) {
-	require_once __DIR__ . '/class-atomic-admin-menu.php';
-	Atomic_Admin_Menu::get_instance();
-} elseif ( ! defined( 'IS_WPCOM' ) || ! IS_WPCOM ) {
+/**
+ * Gets the name of the class that customizes the admin menu.
+ *
+ * @return string Class name.
+ */
+function get_admin_menu_class() {
+	// WordPress.com Atomic sites.
+	if ( jetpack_is_atomic_site() ) {
+		require_once __DIR__ . '/class-atomic-admin-menu.php';
+		return __NAMESPACE__ . '\Atomic_Admin_Menu';
+	}
+
+	// WordPress.com Simple sites.
+	if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+		$blog_id = get_current_blog_id();
+
+		// Domain-only sites.
+		$blog_options   = get_blog_option( $blog_id, 'options' );
+		$is_domain_only = ! empty( $blog_options['is_domain_only'] );
+		if ( $is_domain_only ) {
+			require_once __DIR__ . '/class-domain-only-admin-menu.php';
+			return __NAMESPACE__ . '\Domain_Only_Admin_Menu';
```

Comment:
We can do something like `return Domain_Only_Admin_Menu::class;` to get the FQCN. IDEs offer support for that and makes the navigation a lot easier. :)

Reply:
Nice, much cleaner that way! Done in https://github.com/Automattic/jetpack/pull/19263/commits/32a8e4b1e4a691acea4ba8f772c696add52a0475